frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/InstallNodeAndPnpmMojo.java

```diff
@@ -0,0 +1,120 @@
+package com.github.eirslett.maven.plugins.frontend.mojo;
+
+import com.github.eirslett.maven.plugins.frontend.lib.FrontendPluginFactory;
+import com.github.eirslett.maven.plugins.frontend.lib.InstallationException;
+import com.github.eirslett.maven.plugins.frontend.lib.PNPMInstaller;
+import com.github.eirslett.maven.plugins.frontend.lib.ProxyConfig;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.settings.crypto.SettingsDecrypter;
+import org.apache.maven.settings.Server;
+
+@Mojo(name="install-node-and-pnpm", defaultPhase = LifecyclePhase.GENERATE_RESOURCES, threadSafe = true)
+public final class InstallNodeAndPnpmMojo extends AbstractFrontendMojo {
```

Comment:
This is third class now that also installs `node`... I think it might be time to make ourselves a new base class in-between `InstallNodeAndXMojo` and `AbstractFrontendMojo`. I don't know that we want to do that in this PR, but it feels like there's a wack of duplication now that we could rid ourselves of @eirslett ?

Reply:
Makes sense. My +1 (and also +1 to not do this in this release).